hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -193,6 +195,14 @@ public String getPreCombineField() {
     return props.getProperty(HOODIE_TABLE_PRECOMBINE_FIELD);
   }
 
+  public Option<String[]> getPartitionColumns() {
```

Comment:
why not use an empty array to signify no partition columns?

Reply:
For the previous version of hudi, There are no partition column stored in the hoodie.properties. So It return an `Option#empty` for this case. We can distinguish the case of empty partitions and not store partition columns. I saw other property like `getBootstrapBasePath` also use the Option.